index.html

```diff
@@ -418,6 +419,81 @@ <h2>Transferable MediaStreamTrack</h2>
       <p>The underlying source is supposed to be kept alive between the transfer and transfer-receiving steps, or as long as the data holder is alive.
       In a sense, between these steps, the data holder is attached to the underlying source as if it was a track.</p>
     </div>
+    <h3>MediaStreamTrack frame counters</h3>
+    <div>
+      <p>
+        Frame counters allow you to tell why the frame rate experienced is
+        different from the expected one - for instance, the camera may be
+        slowing down frame production because it's dark, the settings applied
+        may produce decimation of frames, or the system may be CPU starved and
+        therefore dropping frames.
+      </p>
+    </div>
+    <div>
+      <p>
+        The number of non-decimated captured frames can be obtained via the
+        {{MediaStreamTrack/framesCaptured()}} method.
+      </p>
+      <pre class="idl"
+>partial interface MediaStreamTrack {
+    Promise&lt;unsigned long long&gt; framesCaptured();
+};</pre>
+    </div>
+    <section>
+      <h4>Methods</h4>
+      <dl data-link-for="MediaStreamTrack" data-dfn-for="MediaStreamTrack"
+      class="methods">
+        <dt>
+          <dfn>framesCaptured</dfn>
+        </dt>
+        <dd>
+          <p>
+            When this method is called, the user agenst MUST run the following
+            steps:
+          </p>
+          <ol>
+            <li>
+              <p>
+                Let <var>track</var> be the {{MediaStreamTrack}} that this
+                method was called on.
+              </p>
+            </li>
+            <li>
+              <p>Let <var>p</var> be a new promise.</p>
+            </li>
+            <li>
+              <p>
+                In parallel, begin the [= in-parallel steps to get
+                framesCaptured =], given <var>track</var> and <var>p</var>.
+              </p>
+            </li>
+            <li>
+              <p>Return <var>p</var>.</p>
+            </li>
+          </ol>
+          <p>
+            The <dfn>in-parallel steps to get framesCaptured</dfn> given
+            <var>track</var> and <var>p</var> are as follows:
+          </p>
+          <ol>
+            <li>
+              <p>
+                Let <var>framesCaptured</var> be a snapshot of the number of
+                non-decimated captured frames, this includes frames that were
+                dropped by the user agent for any reason other than decimation
+                in order to achieve the configured frame rate settings.
```

Comment:
As I see it, the captured frames would be a value of the track's source.
Which means that if cloning a track and applying different parameters, we would still end up with the same result (modulo race conditions) when calling frameCaptured between two tracks.

Somehow, I would be tempted to add a slot to the track Source.
And then write some text explaining that on each new frame generated by the source and sent to tracks, the counter is incremented. And detail that tracks may decimate frames.

Reply:
Sorry for the late response on this, but I think I liked it better as `getStats`. I think it's unlikely that we'll stop adding stuff (we never do 😉! I mean, timestamp?) — and it scales better (avoids dispatching for each get, which is the only cost here AFAIK). E.g.
```js
const {framesCaptured, somethingElse} = await track.getStats();
```

...is faster (1 vs N dispatches) than
```js
const framesCaptured = await track.getFramesCaptured();
const somethingElse = await track.getSomethingElse();
```

...and simpler than (no-one's gonna bother writing)
```js
const {framesCaptured, someOtherProperty} = await Promise.all([
  track.getFramesCaptured(), track.getSomethingElse()
]);
```
As for a property of the source, a track can have so many different sources, I worry about exposing a single surface for a track's source.

Reply:
More on my reasoning: If we stay with a single get method I'd prefer it be called `getFramesCaptured` for consistency with our other get methods. And once we have a getMethod, might as well return a dictionary. That's how I arrived back at `getStats`, which in my mind would simply return a dictionary, unlike in WebRTC.

Re timestamp: would you expect a timestamp if the API were `const ms = track.framesCaptured`? If not, then it shouldn't be necessary for `const {ms} = await track.getStats()` either IMHO, because the only timing difference is that in the former case the dispatch is on some regular cadence, whereas in the latter case it's on demand. So if anything, the latter sample is going to be more more accurate to right now, not less.